Framework/Core/include/Framework/Expressions.h

```diff
@@ -374,49 +370,49 @@ inline Node ifnode(C condition_, T then_, E else_)
 }
 
 template <>
-inline Node ifnode(Node condition_, Node then_, Node else_)
+inline Node ifnode(Node&& condition_, Node&& then_, Node&& else_)
 {
   return Node{ConditionalNode{}, std::move(then_), std::move(else_), std::move(condition_)};
 }
 
-template <typename L, std::enable_if_t<std::is_integral<L>::value || std::is_floating_point<L>::value, bool> = true>
-inline Node ifnode(Node condition_, Node then_, L else_)
+template <typename L>
+inline Node ifnode(Node&& condition_, Node&& then_, L else_) requires(std::is_integral<L>::value || std::is_floating_point<L>::value)
 {
   return Node{ConditionalNode{}, std::move(then_), LiteralNode{else_}, std::move(condition_)};
 }
 
-template <typename L, std::enable_if_t<std::is_integral<L>::value || std::is_floating_point<L>::value, bool> = true>
-inline Node ifnode(Node condition_, L then_, Node else_)
+template <typename L>
+inline Node ifnode(Node&& condition_, L then_, Node&& else_) requires(std::is_integral<L>::value || std::is_floating_point<L>::value)
 {
   return Node{ConditionalNode{}, LiteralNode{then_}, std::move(else_), std::move(condition_)};
 }
 
-template <typename L1, typename L2, std::enable_if_t<(std::is_integral<L1>::value || std::is_floating_point<L1>::value) && (std::is_integral<L2>::value || std::is_floating_point<L2>::value), bool> = true>
-inline Node ifnode(Node condition_, L1 then_, L2 else_)
+template <typename L1, typename L2>
+inline Node ifnode(Node&& condition_, L1 then_, L2 else_) requires((std::is_integral<L1>::value || std::is_floating_point<L1>::value) && (std::is_integral<L2>::value || std::is_floating_point<L2>::value))
 {
   return Node{ConditionalNode{}, LiteralNode{then_}, LiteralNode{else_}, std::move(condition_)};
 }
 
 template <typename T>
-inline Node ifnode(Configurable<T> condition_, Node then_, Node else_)
+inline Node ifnode(Configurable<T> const& condition_, Node&& then_, Node&& else_)
```

Comment:
do you still need the move?

Reply:
Probably not, I should replace it with forward.

Reply:
I think you do not even need that, assuming Node has the move constructor, no?

Reply:
Node contains unique pointer, so it only has the move constructor. Forward should ensure it is only used once, though.